hudi-common/src/test/java/org/apache/hudi/common/util/TestRetryHelper.java

```diff
@@ -49,4 +51,19 @@ public void testCheckIfExceptionInRetryList() throws Exception {
     retry = (boolean) privateOne.invoke(retryHelper, new UnsupportedOperationException("test"));
     assertTrue(retry);
   }
+
+  @Test
+  public void testCheckTooManyTimes() {
+    int maxRetries = 100;
+    RetryHelper retryHelper = new RetryHelper(INTERVAL_TIME, maxRetries, INTERVAL_TIME, null);
+    AtomicInteger counter = new AtomicInteger(0);
+    assertDoesNotThrow(() -> {
+      retryHelper.start(() -> {
```

Comment:
100 times retry does not really reach the upper threshold `LONG.MAX_VALUE`. did you encounter the overflow in production?

Reply:
> 100 times retry does not really reach the upper threshold `LONG.MAX_VALUE`. 
2^100 > 2^63-1 (LONG.MAX_VALUE)
> did you encounter the overflow in production?
Yes! It fails our writer task.

Reply:
but does sleeping with `Long.MAX_VALUE` makes sense in produciton?

Reply:
> Maybe we should have another explicit upper threshold instead.

There is already code logic that specifies this upper limit.
<img width="549" alt="image" src="https://github.com/user-attachments/assets/f990ebcf-1f08-4c2a-b45a-aff4aab15531">
`getWaitTimeExp` just need to calculate expect max-wait-time (also need to avoid long overflow), real-wait-time decides on `Math.min(expect-max-wait-time, maxIntervalTime)`, `maxIntervalTime` can be configured.